drivers/sensor/nrf5/temp_nrf5.c

```diff
@@ -95,18 +96,10 @@ static const struct sensor_driver_api temp_nrf5_driver_api = {
 	.channel_get = temp_nrf5_channel_get,
 };
 
-static int temp_nrf5_init(struct device *dev);
-
-static struct temp_nrf5_data temp_nrf5_driver;
-
-DEVICE_AND_API_INIT(temp_nrf5, CONFIG_TEMP_NRF5_NAME, temp_nrf5_init,
-		    &temp_nrf5_driver, NULL,
-		    POST_KERNEL,
-		    CONFIG_SENSOR_INIT_PRIORITY, &temp_nrf5_driver_api);
+DEVICE_DECLARE(temp_nrf5);
```

Comment:
Introduced DEVICE_DECLARE, and moved DEVICE_AND_API_INIT() to the bottom to match other implementations (such as UARTE, and HW PWM drivers)